programs/locker/src/instructions/v3/cancel_vesting_escrow.rs

```diff
@@ -0,0 +1,136 @@
+use anchor_spl::memo::Memo;
+use anchor_spl::token_interface::{
+    close_account, CloseAccount, Mint, TokenAccount, TokenInterface,
+};
+use util::{
+    harvest_fees, parse_remaining_accounts, transfer_to_user_v3, AccountsType,
+    ParsedRemainingAccounts, TRANSFER_MEMO_CANCEL_VESTING,
+};
+use crate::util::MemoTransferContext;
+use crate::*;
+
+/// Accounts for [locker::cancel_vesting_escrow_v3].
+#[derive(Accounts)]
+#[event_cpi]
+pub struct CancelVestingEscrowV3<'info> {
+    /// Escrow.
+    #[account(
+        mut,
+        has_one = token_mint,
+        constraint = escrow.load()?.cancelled_at == 0 @ LockerError::AlreadyCancelled
+    )]
+    pub escrow: AccountLoader<'info, VestingEscrowV3>,
+
+    /// Mint.
+    #[account(mut)]
+    pub token_mint: Box<InterfaceAccount<'info, Mint>>,
+
+    /// Escrow Token Account.
+    #[account(
+        mut,
+        associated_token::mint = token_mint,
+        associated_token::authority = escrow,
+        associated_token::token_program = token_program
+    )]
+    pub escrow_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// Creator Token Account.
+    #[account(
+        mut,
+        associated_token::mint = token_mint,
+        associated_token::authority = escrow.load()?.creator,
+        associated_token::token_program = token_program
+    )]
+    pub creator_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// CHECKED: The Token Account will receive the rent
+    #[account(mut)]
+    pub rent_receiver: UncheckedAccount<'info>,
+
+    /// Signer.
+    #[account(mut)]
+    pub signer: Signer<'info>,
```

Comment:
why there is no check for signer, it means anyone can cancel vesting escrow, while only creator can do this function